autogen.pl

```diff
@@ -1178,7 +1178,7 @@ sub in_tarball {
 dnl This file is automatically created by autogen.pl; it should not
 dnl be edited by hand!!
 dnl
-dnl Generated by $username at " . localtime(time) . "
+dnl Generated by $username at " . localtime($ENV{SOURCE_DATE_EPOCH} || time) . "
```

Comment:
I don't think that this does what you expect.  My reading of https://wiki.debian.org/ReproducibleBuilds/TimestampsProposal#Bash_.2F_POSIX_shell is that it expects `SOURCE_DATE_EPOCH` to be of the form 'YYYY-MM-DD'.  If you pass a string of the form 'YYYY-MM-DD' to `localtime()`, you get... weirdness:

```
$ perl -e 'print localtime("2010-10-10") . "\n"';
Wed Dec 31 16:33:30 1969
$ perl -e 'print localtime("2018-10-10") . "\n"';
Wed Dec 31 16:33:38 1969
```

Did you mean:

```
... . ($ENV{SOURCE_DATE_EPOCH} || localtime(time)) . ...
```

Reply:
no, it is a plain int. In the linked example, you can see `${SOURCE_DATE_EPOCH:-$(date +%s)}`